base copy for Unreal + adding references everywhere

pasotee opened the pull request:
### Description

First iteration for the Unreal Engine SDK docs.

### Requirement checklist

- [X] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

kp-cat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,720 @@
+| Properties              | Description                                                                                                                                                                                                                                                                                         |
+| ----------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `SdkKey`                | SDK Key to access your feature flags and configurations. Get it from ConfigCat Dashboard.                                                                                                                                                                                                           |
+| `BaseUrl`               | Optional, sets the CDN base url (forward proxy, dedicated subscription) from where the sdk will download the configurations.                                                                                                                                                                        |
```
```suggestion
| `BaseUrl`               | Optional, sets the CDN base url (forward proxy, dedicated subscription) from where the sdk will download the config JSON.                                                                                                                                                                        |
```

kp-cat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+| `ConnectTimeoutMs`      | Optional, defaults to `8000ms`. Sets the amount of milliseconds to wait for the server to make the initial connection (i.e. completing the TCP connection handshake). `0` means it never times out during transfer                                                                                  |
+| `ReadTimeoutMs`         | Optional, defaults to `5000ms`. Sets the amount of milliseconds to wait for the server to respond before giving up. `0` means it never times out during transfer.                                                                                                                                   |
+| `PollingMode`           | Optional, sets the polling mode for the client. [More about polling modes](#polling-modes).                                                                                                                                                                                                         |
+| `AutoPollInterval`      | For PollingMode == Custom, sets at least how often this policy should fetch the latest configuration and refresh the cache.                                                                                                                                                                         |
```
```suggestion
| `AutoPollInterval`      | For PollingMode == Custom, sets at least how often this policy should fetch the latest config JSON and refresh the cache.                                                                                                                                                                         |
```

pasotee replied:
I missed this suggestion, I will apply the change manually next commit.

laliconfigcat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+| ----------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Sdk Key`                     | SDK Key to access your feature flags and configurations. Get it from ConfigCat Dashboard.                                                                                                                                                                                                           |
+| `Base Url`                    | Optional, sets the CDN base url (forward proxy, dedicated subscription) from where the sdk will download the config JSON.                                                                                                                                                                        |
+| `Data Governance`             | Optional, defaults to `Global`. Describes the location of your feature flag and setting data within the ConfigCat CDN. This parameter needs to be in sync with your Data Governance preferences. [More about Data Governance](advanced/data-governance.md). Available options: `Global`, `EuOnly`.  |
```
```suggestion
| `Data Governance`             | Optional, defaults to `Global`. Describes the location of your feature flag and setting data within the ConfigCat CDN. This parameter needs to be in sync with your Data Governance preferences. [More about Data Governance](/advanced/data-governance). Available options: `Global`, `EuOnly`.  |
```
The internal link format changed a bit, and the correct ones should start with a / and not end with .md

laliconfigcat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+| --------------- | ------------------------------------------------------------------------------------------------------------ |
+| `Key`           | **REQUIRED.** The key of a specific setting or feature flag. Set on _ConfigCat Dashboard_ for each setting.  |
+| `Default Value` | **REQUIRED.** This value will be returned in case of an error.                                               |
+| `User`          | Optional, _User Object_. Essential when using Targeting. [Read more about Targeting.](advanced/targeting.md) |
```
```suggestion
| `User`          | Optional, _User Object_. Essential when using Targeting. [Read more about Targeting.](/advanced/targeting) |
```

laliconfigcat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+## User Object
+
+The [User Object](../advanced/user-object.md) is essential if you'd like to use ConfigCat's [Targeting](advanced/targeting.md) feature.
```
```suggestion
The [User Object](/advanced/user-object) is essential if you'd like to use ConfigCat's [Targeting](/advanced/targeting) feature.
```

laliconfigcat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+You can download your current config JSON from ConfigCat's CDN and use it as a baseline.
+
+The URL to your current config JSON is based on your [Data Governance](advanced/data-governance.md) settings:
```
```suggestion
The URL to your current config JSON is based on your [Data Governance](/advanced/data-governance) settings:
```

laliconfigcat reviewed (changes requested):
Looks good to me, but we had some changes regarding internal links. Could you please change them?

pasotee commented:
Sure thing. I've committed the suggested edits and fixed one more link I found locally using the old format.

sigewuzhere commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+Via **[Unreal Marketplace](https://www.unrealengine.com/marketplace/product/e142293a397d4ce4bf6a1f3053a2316d)**
+
+Via **[Github clone](https://github.com/configcat/unreal-engine-sdk)**
```
```suggestion
Via **[GitHub clone](https://github.com/configcat/unreal-engine-sdk)**
```

sigewuzhere commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+## Custom Cache
+
+**Coming soon**: [https://github.com/configcat/unreal-engine-sdk/issues/9](Support for Custom Cache #9)
```
<img width="912" alt="image" src="https://github.com/configcat/docs/assets/58779216/8366afe3-1e59-480f-b469-60c69c15728e">

Improperly formatted link

sigewuzhere commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+The verbosity can be changed:
+
+Via [Command line arguments]:
```
```suggestion
Via **Command line arguments**:
```

sigewuzhere commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+Run the executable with `-LogCmds="LogConfigCat VerbosityLevel"`. For example: `-LogCmds="LogConfigCat Warning"` to only show warnings and above.
+
+Via [DefaultEngine.ini]:
```
```suggestion
Via **DefaultEngine.ini**:
```

sigewuzhere commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+
+## Sensitive information handling
+
+The frontend/mobile SDKs are running in your users' browsers/devices. The SDK is downloading a [config JSON](/requests/) file from ConfigCat's CDN servers. The URL path for this config JSON file contains your SDK key, so the SDK key and the content of your config JSON file (feature flag keys, feature flag values, targeting rules, % rules) can be visible to your users.
```
```suggestion
The frontend/mobile SDKs are running in your users' browsers/devices. The SDK is [downloading a config JSON](/requests/) file from ConfigCat's CDN servers. The URL path for this config JSON file contains your SDK key, so the SDK key and the content of your config JSON file (feature flag keys, feature flag values, targeting rules, % rules) can be visible to your users.
```

sigewuzhere reviewed (changes requested):
Hey @pasotee !
A few links are missing from:
- sidebars.js
- index.js

Something is wrong with the header:
<img width="1031" alt="image" src="https://github.com/configcat/docs/assets/58779216/f3c2c0a6-5df1-4f52-bd1a-cf14a47ef1bc">

pasotee commented:
hey @sigewuzhere, thanks for taking a look. I've applied your suggestions and fixed the broken links.

Regarding the links (sidebar.js & index.js), I've removed them on purpose. I will submit the SDK for approval from the Unreal team (they will use a static link to access to documentation during the review process) and once it's publicly available, I will make a follow-up PR to add the missing links.

Regarding the broken header, it looks like there is an extra `n` at the start of the markdown file like this:

```
---
id: unreal
title: Unreal Engine SDK Reference
description: ConfigCat Unreal Engine SDK Reference. This is ...
---
```

to 

```
n---
id: unreal
title: Unreal Engine SDK Reference
description: ConfigCat Unreal Engine SDK Reference. This is ...
---
```

I don't see it on the GitHub version, can you confirm if this is a local problem? Otherwise I will try to dig deeper into it.

kp-cat commented on website/docs/sdk-reference/unreal.md:
```diff
@@ -0,0 +1,712 @@
+| `Connect Timeout`             | Optional, defaults to `8000ms`. Sets the amount of milliseconds to wait for the server to make the initial connection (i.e. completing the TCP connection handshake). `0` means it never times out during transfer                                                                                  |
+| `Read Timeout`                | Optional, defaults to `5000ms`. Sets the amount of milliseconds to wait for the server to respond before giving up. `0` means it never times out during transfer.                                                                                                                                   |
+| `Polling Mode`                | Optional, sets the polling mode for the client. [More about polling modes](#polling-modes).                                                                                                                                                                                                         |
+| `Auto Poll Interval`          | For PollingMode == Custom, sets at least how often this policy should fetch the latest config JSO and refresh the cache.                                                                                                                                                                            |
```
```suggestion
| `Auto Poll Interval`          | For PollingMode == Custom, sets at least how often this policy should fetch the latest config JSON and refresh the cache.                                                                                                                                                                           |
```

sigewuzhere reviewed (approved):
Hi @pasotee ! Thanks, yeah the n was a local mistake, sorry about that. All good to go from my end.
